browser/src/core/Socket.js

```diff
@@ -1557,6 +1557,7 @@ app.definitions.Socket = L.Class.extend({
 				this._map._isCursorVisible = false;
 
 			this._map._docLayer._resetCanonicalIdStatus();
+			this._map.uiManager.previousTheme = null;
```

Comment:
If _resetCanonicalIdStatus sets this._canonicalIdInitialized to false I then see that we have:

if (!this._canonicalIdInitialized) {
        this._canonicalIdInitialized = true;
        this._update();
} else if (viewRenderedState !== this._map.uiManager.previousTheme) {
        this._requestNewTiles();
        this._invalidateAllPreviews();
        this.redraw();
}

inside else if (textMsg.startsWith('canonicalidchange:')) 

does that mean we get two canonicalidchange messages on a reconnect?, one setting this._canonicalIdInitialized back to true, and then a second one which triggers the else this._requestNewTiles ... ?

Reply:
Yes we get two canonicalidchange message when we load/reconnect

Reply:
This might be interesting
https://github.com/CollaboraOnline/online/pull/7324